crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -279,66 +295,343 @@ where
         Ok(())
     }
 
+    /// Look up the IBC denomination from a IbcToken.
```

Comment:
Should this function be defined in the `ibc` crate instead, and imported here? Seems like we might want to use it in multiple places (and I wonder if it is in fact already defined somewhere...)

Reply:
You're right, this function is already implemented in the client (where the queries are sent to an RPC node). I'll investigate whether the protocol version here (which reads storage directly) can somehow reuse this logic.

Reply:
I've discussed turning this function into an RPC endpoint with @tzemanovic . Doing so seems suboptimal because this function is a essentially a heuristic for guessing IBC denominations from `IbcToken` hashes based on the results the protocol has cached so far. Also it seems to me that somehow using generics to share this logic would be overkill for this limited issue. Do you think there's a way to eliminate the use of this function in the MASP VP @yito88 ?